docs/DOCUMENTATION_GUIDELINES.md

```diff
@@ -0,0 +1,254 @@
+
+### Top Level Pallet Docs (`lib.rs`)
+
+For the top-level pallet docs, consider the following template:
+
+```
+//! # <Pallet Name>
+//!
+//! <single-liner about the pallet>.
+//!
+//! ## Overview
+//!
+//! <should be high-level details that are relevant to the most broad audience>
+//!
+//! <The audience here is potentially non-coders who just want to know what this pallet does, not how it does it>
+//!
+//! <potentially a few paragraphs, focus on what external folks should know about the pallet>
+//!
+//! ### Example
+//!
+//! <Your pallet must have a few tests that cover important user journeys. Use https://crates.io/crates/docify to reuse these as examples>.
+//!
+//! ## Pallet API
+//!
+//! <Reminder: inside the [`pallet`] module, a template that leads the reader to the relevant items is auto-generated. There is no need to repeat things like "See Config trait for ...", which are generated inside [`pallet`] here anyways. You can use the below line as-is:>
+//!
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes, including its configuration trait, dispatchables, storage items, events and errors.
+//!
+//! <The audience of this is those who want to know how this pallet works, to the extent of being able to build something on top of it, like a DApp or another pallet>
+//!
+//! This section can most often be left as-is.
+//!
+//! ## Low Level / Implementation Details
+//!
+//! <The format of this section is up to you, but we suggest the Design-oriented approach that follows>
+//!
+//! <The audience of this would be your future self, or anyone who wants to gain a deep understanding of how the pallet works so that they can eventually propose optimizations to it>
+//!
+//! ### Design Goals (optional)
+//!
+//! <Describe your goals with the pallet design.>
+//!
+//! ### Design (optional)
+//!
+//! <Describe how you've reached those goals. This should describe the storage layout of your pallet and what was your approach in designing it that way.>
+//!
+//! ### Terminology (optional)
+//!
+//! <Optionally, explain any non-obvious terminology here. You can link to it if you want to use the terminology further up>
+```
+
+This template's details (heading 3s and beyond) are left flexible, and at the discretion of the developer to make the best final choice about. For example, you might want to include `### Terminology` or not. Moreover, you might find it more useful to include it in `## Overview`.
+
+Nonetheless, the high level flow of going from the most high level explanation to the most low level explanation is important to follow.
+
+As a rule of thumb, the Heading 2s (`##`) in this template can be considered a strict rule, while the Heading 3s (`###`) and beyond are flexible.
+
+#### Polkadot and Substrate
+
+Optionally, in order to demonstrate the relation between the two, you can start the pallet documentation with:
+
+```
+//! > Made with *Substrate*, for *Polkadot*.
```

Comment:
sounds a bit exclusive, when in reality used not only by Polkadot.

Reply:
well to be fair "made for" doesn't mean it couldn't be used for other things, but this is a good point

Reply:
This is a trial just for marketing to see. We are still yet to decide if we want to add this in all pallets and crates.